src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,6 +39,8 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        String[] splited = trimmedAddress.split(SEPARATION);
+        putInPlace(splited);
```

Comment:
Good attempt to maintain SLAP here. Can consider putting the trimming + split + storage together in a new method instead of just the storage part.